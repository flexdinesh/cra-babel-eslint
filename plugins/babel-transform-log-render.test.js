import { transformSync } from "@babel/core";
import logRenderPlugin from "./babel-transform-log-render";

const HelloWorld = `
const HelloWorld = () => {
  const [val, setVal] = useState(true);
  const [anotherVal, setAnotherVal] = useState(100);
  return <span>Hello World</span>;
};
`;

test("inject console.log", () => {
  const { code } = transformSync(HelloWorld, {
    ast: true,
    plugins: ["@babel/plugin-transform-react-jsx", logRenderPlugin]
  });

  expect(code).toMatchSnapshot();
});
