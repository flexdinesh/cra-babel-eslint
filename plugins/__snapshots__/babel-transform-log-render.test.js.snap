// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inject console.log 1`] = `
"const HelloWorld = () => {
  console.log(\\"%c >> RenderLog >> \\", \\"color: hotpink;\\", \\"HelloWorld\\");
  const [val, setVal] = useState(true);
  const [anotherVal, setAnotherVal] = useState(100);
  return React.createElement(\\"span\\", null, \\"Hello World\\");
};"
`;
